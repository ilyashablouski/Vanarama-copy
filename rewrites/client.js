/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
