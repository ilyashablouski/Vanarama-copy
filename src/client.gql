extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
