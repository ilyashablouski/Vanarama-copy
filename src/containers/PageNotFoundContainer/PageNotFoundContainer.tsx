import React from 'react';
import { NextPage } from 'next';
import dynamic from 'next/dynamic';
import ReactMarkdown from 'react-markdown';
import Breadcrumbs from 'core/atoms/breadcrumbs-v2';
import { IServiceBanner } from 'core/molecules/service-banner/interfaces';
import ServiceBanner from 'core/molecules/service-banner';
import React from 'react';
import RouterLink from '../../components/RouterLink/RouterLink';
import {
  GenericPageQuery_genericPage_sections_cards_cards,
  GenericPageQuery_genericPage_sections_featured,
} from '../../../generated/GenericPageQuery';
import Skeleton from '../../components/Skeleton';
import SectionCards from '../../components/SectionCards';
import { Nullish } from '../../types/common';
import { IBreadcrumbLink } from '../../types/breadcrumbs';

const Heading = dynamic(() => import('core/atoms/heading'), {
  loading: () => <Skeleton count={2} />,
});
const Text = dynamic(() => import('core/atoms/text'), {
  loading: () => <Skeleton count={1} />,
});
const ImageV2 = dynamic(() => import('core/atoms/image/ImageV2'), {
  loading: () => <Skeleton count={3} />,
});

interface IProps {
  name: string | null | undefined;
  cards?:
    | (GenericPageQuery_genericPage_sections_cards_cards | null)[]
    | null
    | undefined;
  breadcrumbsItems?: Nullish<IBreadcrumbLink[]>;
  featured?: GenericPageQuery_genericPage_sections_featured | null | undefined;
  serviceBanner?: IServiceBanner;
}

const PageNotFoundContainer: NextPage<IProps> = ({
  name,
  featured,
  breadcrumbsItems,
  cards,
  serviceBanner,
}) => {
  return (
    <>
      <ServiceBanner
        enable={serviceBanner?.enable}
        message={serviceBanner?.message}
        link={serviceBanner?.link}
        className="-mb-500"
      />
      <div className="row:title">
        <Breadcrumbs items={breadcrumbsItems} />
        <Heading
          tag="h1"
          size="xlarge"
          color="black"
          dataUiTestId="not-found-page_heading"
        >
          {name || ''}
        </Heading>
      </div>
      <div className="row:featured-left">
        <ImageV2
          quality={60}
          optimisedHost
          lazyLoad={false}
          className="-white"
          width={featured?.image?.file?.details.image.width}
          height={featured?.image?.file?.details.image.height}
          src={featured?.image?.file?.url || ''}
          size="expand"
        />
        <div className="full-height">
          <ReactMarkdown
            allowDangerousHtml
            source={featured?.body || ''}
            renderers={{
              paragraph: props => (
                <Text {...props} size="lead" tag="p" color="darker" />
              ),
              link: props => {
                const { href, children } = props;
                return (
                  <RouterLink
                    link={{ href, label: children }}
                    classNames={{ color: 'teal' }}
                  />
                );
              },
            }}
          />
        </div>
      </div>
      <div className="row:bg-lighter -col-300">
        <div className="row:cards-3col">
          {cards && <SectionCards cards={cards} />}
        </div>
      </div>
    </>
  );
};

export default PageNotFoundContainer;
