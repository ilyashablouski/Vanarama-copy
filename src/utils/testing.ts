// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
