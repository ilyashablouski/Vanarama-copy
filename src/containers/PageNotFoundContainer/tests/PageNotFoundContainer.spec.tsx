import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { screen, render, waitFor } from '@testing-library/react';
import PageNotFoundContainer from '../PageNotFoundContainer';

jest.mock('../../../hooks/useMediaQuery');
jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
  }),
}));

// ARRANGE
const METADATA = {
  canonicalUrl: 'https://www.vanarama.com/car-leasing.html',
  legacyUrl: 'https://www.vanarama.com/car-leasing.html',
  metaDescription:
    'Find unbeatable Car Leasing Deals at Vanarama. Get top personal & business lease offers on brand new, in-stock cars in every make and model. Save money and lease your dream car today.',
  metaRobots: 'all',
  name: 'HubCarPage',
  pageType: null,
  publishedOn: null,
  slug: 'hubcarpage',
  title: 'Car Leasing Deals | Personal & Business Contract Hire | Vanarama',
  schema: null,
  breadcrumbs: null,
};

const NAME = 'Top 5 Electric Vans Sort Of';

const CARDS = [
  {
    body:
      'Find the perfect van lease for your business with a huge range of brand new, in-stock, commercial vehicles to choose from at unbeatable prices.',
    image: {
      description: 'Trafic',
      file: {
        fileName: 'renaulttrafic0917(2).jpg',
        url:
          '//images.ctfassets.net/3xid768u5joa/22zuZfMLYzskVi3MQPcu7c/32caa929339dc2e88eee09ea2af84b79/renaulttrafic0917_2_.jpg',
      },
      title: 'Renaulttrafic0917(2)-full',
    },
    link: {
      text: 'Search Vans',
      url: '/van-leasing.html',
    },
    name: 'Search Vans',
    title: 'Search Vans',
    titleTag: 'h2',
  },
  {
    body:
      'Drive a brand new car with affordable monthly payments. We offer superb business & personal contract hire deals on a range of lease cars.',
    image: {
      description: 'Trafics',
      file: {
        fileName: 'renaulttrafic0917(2).jpg',
        url:
          '//images.ctfassets.net/3xid768u5joa/22zuZfMLYzskVi3MQPcu7c/32caa929339dc2e88eee09ea2af84b79/renaulttrafic0917_2_.jpg',
      },
      title: 'Renaulttrafic0917(2)-full',
    },
    link: {
      text: 'Search Cars',
      url: '/van-leasing.html',
    },
    name: 'Search Cars',
    title: 'Search Cars',
    titleTag: 'h2',
  },
];

const FEATURED = {
  body:
    "It looks like you've ventured down the wrong road. Let's get you back on track. Just choose your preferred vehicle to get you back where you should be ...",
  cards: null,
  iconList: null,
  image: {
    description: null,
    file: {
      fileName: 'shutterstock_773667343.jpg',
      url:
        '//images.ctfassets.net/3xid768u5joa/3onM1VAXZCIRTHPcHbfD55/f24511517067f91e7008765ff0c9656a/shutterstock_773667343.jpg',
    },
    title: 'shutterstock 773667343',
  },
  layout: ['Media Left'],
  link: null,
  testimonials: null,
  title: null,
  titleTag: null,
  video: null,
};

describe('<PageNotFoundContainer />', () => {
  it('should match snapshot', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider addTypename={false}>
        <PageNotFoundContainer
          cards={CARDS}
          name={NAME}
          featured={FEATURED}
          breadcrumbsItems={null}
          metaData={METADATA}
        />
      </MockedProvider>,
    );
    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(`Top 5 Electric Vans Sort Of`),
      ).toBeInTheDocument();
    });

    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
