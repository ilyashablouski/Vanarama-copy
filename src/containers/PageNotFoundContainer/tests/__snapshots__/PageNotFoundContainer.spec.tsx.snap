// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFoundContainer /> should match snapshot 1`] = `
<body>
  <div>
    <div
      class="row:title"
    >
      <h1
        class="heading -black -xlarge"
      >
        Top 5 Electric Vans Sort Of
      </h1>
    </div>
    <div
      class="row:featured-left"
    >
      <div
        class="lazyload-wrapper"
      >
        <div
          class="lazyload-wrapper"
        >
          <div
            class="lazyload-placeholder"
          />
        </div>
      </div>
      <div
        class="full-height"
      >
        <p
          class="text -lead -darker"
        >
          It looks like you've ventured down the wrong road. Let's get you back on track. Just choose your preferred vehicle to get you back where you should be ...
        </p>
      </div>
    </div>
    <div
      class="row:bg-lighter -col-300"
    >
      <div
        class="row:cards-3col"
      >
        <div
          class="card"
          data-testid="card"
        >
          <div
            class="card-image-wrapper"
          >
            <div
              class="card-image-placeholder"
            />
          </div>
          <div
            class="title -flex-h"
          >
            <a
              class="heading -black -lead"
              data-testid="link"
              href="/van-leasing.html"
              rel="noopener noreferrer"
            >
              Search Vans
            </a>
            <button
              class="button -teal -xsmall -solid -round"
            >
              <div
                class="button--inner"
              >
                <i
                  class="icon -regular md hydrated -white"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <title>
                      ionicons-v5-a
                    </title>
                    <polyline
                      points="268 112 412 256 268 400"
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                    />
                    <line
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                      x1="392"
                      x2="100"
                      y1="256"
                      y2="256"
                    />
                  </svg>
                </i>
              </div>
            </button>
          </div>
          <span
            class="text -regular -dark"
            data-testid="card-description"
          >
            Find the perfect van lease for your business with a huge range of brand new, in-stock, commercial vehicles to choose from at unbeatable prices.
          </span>
        </div>
        <div
          class="card"
          data-testid="card"
        >
          <div
            class="card-image-wrapper"
          >
            <div
              class="card-image-placeholder"
            />
          </div>
          <div
            class="title -flex-h"
          >
            <a
              class="heading -black -lead"
              data-testid="link"
              href="/van-leasing.html"
              rel="noopener noreferrer"
            >
              Search Cars
            </a>
            <button
              class="button -teal -xsmall -solid -round"
            >
              <div
                class="button--inner"
              >
                <i
                  class="icon -regular md hydrated -white"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <title>
                      ionicons-v5-a
                    </title>
                    <polyline
                      points="268 112 412 256 268 400"
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                    />
                    <line
                      style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                      x1="392"
                      x2="100"
                      y1="256"
                      y2="256"
                    />
                  </svg>
                </i>
              </div>
            </button>
          </div>
          <span
            class="text -regular -dark"
            data-testid="card-description"
          >
            Drive a brand new car with affordable monthly payments. We offer superb business & personal contract hire deals on a range of lease cars.
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
